### PR TITLE
Issue3063 District DHW

### PR DESCRIPTION
**Pull request by dhblum**
This is for #3063.  In particular, it:

- Implements component level domestic hot water models for two sources: 1) district heat exchanger and auxiliary electric heat and 2) water-to-water heat pump with storage tank.
- Implements component level domestic hot water models for fixture load(s) and thermostatic mixing valve.
- Adds a building load option using the Building Time Series load model and heat pump-based ETS model that includes the heat pump with storage tank DHW generation source.  Led to creation of associated partial ETS class that is extended with the original DHW modeling approach in one model, and the new one in another.
- Various updates to model names and documentation.
- Commits unit test results.

Still todo:

- [x] Review
- [x] Update release notes 
- [x] Check openmodelica unit tests pass with 1.22 update
- [x] Add optimica unit tests to exclusion list

### CODE REVIEW

File: Buildings/Experimental/DHC/Loads/HotWater/ThermostaticMixingValve.mo
Comment (mwetter) [inline Buildings/Experimental/DHC/Loads/HotWater/ThermostaticMixingValve.mo:56]: Does this controller really need a reset, and hence an associated hysteresis block that triggers events?

**Review by mwetter — COMMENTED** [review]
See inline comments for initial feedback.

File: Buildings/Resources/Data/Experimental/DHC/Loads/Heating/DHW/DHW_SingleApartment.mos
Comment (mwetter) [inline Buildings/Resources/Data/Experimental/DHC/Loads/Heating/DHW/DHW_SingleApartment.mos:1]: This file is quite large, and there is no citation or explanation of where it comes from, hence I don't see how a user would use it. Consider shortening the file, adding a source, and/or replacing it with a generation of a load profile.

**Comment by dhblum** [comment]
@mwetter I've made some updates to the models we discussed as well as updated documentation, updated some example models, removed old code and cleanup, committed unit test results, and got example models to simulate in OpenModelica.  There's still an issue I know of where ``Buildings.Experimental.DHC.Loads.Combined.Examples.BuildingTimeSeriesWithETSDHWTank`` fails to simulate in OCT (event iteration fails when DHW turns on).  But this model does simulate in Dymola and OM (locally).  So I need to look into this a little more.  This should be the only unit test that fails, but I will keep an eye on results if there's others I missed.

In the meantime though, I think this is ready for another review by you and would appreciate it.

**Comment by dhblum** [comment]
The unit test failure at https://app.travis-ci.com/github/lbl-srg/modelica-buildings/jobs/614802226 is an initialization failure for ``Buildings.Experimental.DHC.Loads.Combined.Examples.BuildingTimeSeriesWithETSDHWTank`` with OpenModelica.  However, when simulating locally with OM v 1.22.0, it simulates OK.  So I suspect with the update of the OM docker for Buildings unit testing, this will be fixed.  The other two unit test failures are due to OCT.

**Comment by dhblum** [comment]
@mwetter I've updated the port locations and also the icons for the heat pump and heat pump with DHW subsystems.  As far as I know, there's still the issue of simulating some models in Optimica.  But let me know if there's anything else needed that you see to make this ready to merge.

**Review by mwetter — APPROVED** [review]
@dhblum : Please go ahead and merge it when the release notes are updated (which I think you did today).
Regarding your note about issues with some models in Optimica, I believe these are old issues, right? I am asking as I don't see a new entry in `conf.yml`

**Comment by dhblum** [comment]
Thanks @mwetter.  Yes I added release notes today.  I have not added the optimica errors to conf.yml yet.  The openmodelica tests will also fail if the unit test image isn't updated to 1.22.0.  I can either add to the conf.yml until https://github.com/lbl-srg/modelica-buildings/pull/3603 is merged, or wait for https://github.com/lbl-srg/modelica-buildings/pull/3603 to be merged before merging this one.

**Comment by mwetter** [comment]
@dhblum : I suggest you don't add the openmodelica entries in `conf.yml`. I hope to have fixed the latest timeouts in the PR of https://github.com/lbl-srg/modelica-buildings/pull/3603. The PR should run this morning as there is only one more in the queue ahead.

**Comment by dhblum** [comment]
@mwetter Sounds good, I'll hold off until #3603 is merged.

**Comment by dhblum** [comment]
@mwetter Unit tests for openmodelica are failing still after merging the latest master with update to OM 1.22.0, with a failure of initialization for one model.  But, when I run that model with OM 1.22.0 locally it works fine.  The only difference I can see is that I run on Ubuntu 20 while the travis unit test runs with a Ubuntu 22 (Docker).  Do you have experience that the same OM version but on different OS versions can cause a problem?

**Comment by mwetter** [comment]
Different OS and OS versions may yield slightly different results, such as due to different compiler versions and/or libraries. I suggest you put it on the exclude list for now, and when we update to the new OpenModelica version, revisit it.

**Comment by dhblum** [comment]
Ok I see, thanks.  Sounds reasonable, I'll add it to the exclude list.

File: Buildings/Resources/Data/Experimental/DHC/Loads/Heating/DHW/DHW_SingleApartment.mos
Comment (dhblum) [reply Buildings/Resources/Data/Experimental/DHC/Loads/Heating/DHW/DHW_SingleApartment.mos:1]: This file is removed in https://github.com/lbl-srg/modelica-buildings/pull/3525/commits/3fcbb30dad2060353ebcb9141b4a9fd5cbba84dc and replaced with profile from DOE reference midrise apartment building (https://github.com/lbl-srg/modelica-buildings/pull/3525/files#diff-79fed69cab4962b0d531b3670bebcc90a83aa629113e3094ad4271b43695c69a).

File: Buildings/Experimental/DHC/Loads/HotWater/ThermostaticMixingValve.mo
Comment (dhblum) [reply Buildings/Experimental/DHC/Loads/HotWater/ThermostaticMixingValve.mo]: This has since been refactored.

File: Buildings/Experimental/DHC/Loads/HotWater/HeatPumpWithTank.mo
Comment (dhblum) [reply Buildings/Experimental/DHC/Loads/HotWater/HeatPumpWithTank.mo]: This has since been refactored.

File: Buildings/Experimental/DHC/Loads/HotWater/HeatPumpWithTank.mo
Comment (dhblum) [reply Buildings/Experimental/DHC/Loads/HotWater/HeatPumpWithTank.mo:1]: The model has been refactored to improve these points.

**Comment by dhblum** [comment]
@mwetter When you typically merge to master do you have any preference for squash merges or not?

**Comment by mwetter** [comment]
Either is fine.

**Pull request merged**
